### PR TITLE
Add Broadcast->isOnAir and Service->isActive

### PR DESCRIPTION
**Pull request by BPScott**
Add clarifying comments on if start/end times for services and
broadcasts are inclusive or exclusive (starts are inclusive,
ends are exclusive)

### CODE REVIEW

**Comment by votemike** [comment]
How does this deal with the switch between BST and GMT? E.g. if a broadcast starts at 01:45 and ends at 01:15?

File: src/Domain/Entity/Service.php [hunk 1]
```diff
@@ -125,4 +140,17 @@ public function getLiveStreamUrl(): ?string
...
+
+    /**
+     * Returns true if the service is active on a given date.
+     * If a service has no start date then we assume it was active since the
+     * dawn of time.
+     * If a service has no end date then we assume it will be active until the
+     * end of time.
+     * Service starts are inclusive and ends are exclusive.
+     */
+    public function isActive(DateTimeImmutable $date): bool
```
Comment (votemike) [inline src/Domain/Entity/Service.php]: `isActiveOn`?

**Comment by BjornTwachtmann** [comment]
Logic looks sound, I hope there aren't as many nasty lurking timezone issues in DateTime as in our old code, there shouldn't be really. I agree isActiveOn makes more sense if a date is being passed in.

File: src/Domain/Entity/Broadcast.php [hunk 2]
```diff
@@ -125,4 +137,15 @@ public function isRepeat(): bool
...
         return $this->isRepeat;
     }
+
+    /**
+     * Returns true if the broadcast is on air at a given date.
+     * Broadcast starts are inclusive and ends are exclusive.
+     * Given two broadcasts, the first will end at at 06:30:00 and the second
+     * will start at 06:30:00.
+     */
+    public function isOnAir(DateTimeImmutable $date): bool
```
Comment (votemike) [inline src/Domain/Entity/Broadcast.php]: I know it's fussy as it's only a variable name used internally, but this isn't a date. It's a date/time. So should the variable actually be `$datetime`?  
If this was in a single project, I wouldn't be this fussy, but it's in a composer package that may be used by a number of people and so correctness/readabilty seems important.

**Comment by BPScott** [comment]
All times should be in the UTC timezone, with a TZ signifying it as such. We shall deal with the conversion to UK time (GMT or BST) at render time.

Shall update to `isActiveOn`. Should I change `isOnAir` to `isOnAirOn` for the same reasons? That repetition  of 'on' looks odd, 'isBroadcastingOn`?

Hopefully the move to using`DateTimeImmutable` shall help avoid much of the weird around mutable date instances.

File: src/Domain/Entity/Broadcast.php (hunk 2 above)
Comment (BPScott) [inline src/Domain/Entity/Broadcast.php]: Good spot. Technically correct is the best kind of correct.

**Comment by BPScott** [comment]
Have updated the functions to be `isActiveAt` and `isOnAirAt` and am passing in `$dateTime` to both of them

**Pull request merged**